x-pack/plugins/osquery/server/lib/telemetry/constants.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const TELEMETRY_MAX_BUFFER_SIZE = 100;
```

Comment:
You are going to send 100 items - that is fine. But the total ndjson payload size needs to total less than 10MB.
We don't have a nice way of calculating this at runtime at the moment. If you think this will be greater you should reduce this number. I know queries can get quite big